src/wavelet_matrix.rs

```diff
@@ -0,0 +1,559 @@
+use anyhow::{anyhow, Result};
+use std::ops::Range;
+
+use crate::{broadword, BitVector, RsBitVector};
+
+/// Time and space efficient index data structures for sequence.
+///
+/// [`WaveletMatrix`] implements stores a sequence of integers and provides myriad operations on sequece.
+/// When a sequence stores $`n`$ integers from $`[0, u-1]`$,
+/// most of the operations run in $`O(log u)`$ , using  $`n log u + o (n log u) + O(log u log n)`$ bits.
+///
+/// This is a yet another Rust port of [hillbig's waveletMatrix](https://github.com/hillbig/waveletTree/blob/master/waveletMatrix.go).
+///
+/// # Example
+///
+/// ```
+/// use sucds::{WaveletMatrix, WaveletMatrixBuilder};
+///
+/// let mut wmb = WaveletMatrixBuilder::new();
+/// let s = "tobeornottobethatisthequestion";
+/// s.chars().for_each(|c| wmb.push(c as usize));
+///
+/// let wm = wmb.build().unwrap();
+/// assert_eq!(wm.num(), s.len());
+/// assert_eq!(wm.dim(), ('u' as usize) + 1);
+///
+/// assert_eq!(wm.lookup(20), 'h' as usize);
+/// assert_eq!(wm.rank(22, 'o' as usize), 4);
+/// assert_eq!(wm.select(2, 't' as usize), 9);
+/// ```
+///
+/// # References
+///
+/// - F. Claude, and G. Navarro, "The Wavelet Matrix," In SPIRE 2012.
+#[derive(Default, Debug, PartialEq, Eq)]
+pub struct WaveletMatrix {
+    layers: Vec<RsBitVector>,
+    dim: usize,
+    num: usize,
```

Comment:
```suggestion
    len: usize,
```

Rust often uses `len` to indicate the size of data. Could you modify `num` into `len`?

Reply:
fixed in 40d3c90